src/inmanta/util.py

```diff
@@ -112,31 +115,130 @@ def handler(future):
     return future
 
 
-TaskMethod = Callable[[], Awaitable[None]]
+TaskMethod = Callable[[], Awaitable[object]]
+
+
+class TaskSchedule(ABC):
+    """
+    Abstract base class for a task schedule specification. Offers methods to inspect when the task should be scheduled, relative
+    to the current time. Stateless.
+    """
+
+    @abstractmethod
+    def get_initial_delay(self) -> float:
+        """
+        Returns the number of seconds from now until this task should be scheduled for the first time.
+        """
+        ...
+
+    @abstractmethod
+    def get_next_delay(self) -> float:
+        """
+        Returns the number of seconds from now until this task should be scheduled again. Uses the current time as reference,
+        i.e. assumes the task has already run at least once and a negligible amount of time has passed since the last run
+        completed.
+        """
+        ...
+
+    @abstractmethod
+    def log(self, action: TaskMethod) -> None:
+        """
+        Log a message about the action being scheduled according to this schedule.
+        """
+        ...
+
+
+ScheduledTask = Tuple[TaskMethod, TaskSchedule]
```

Comment:
Not really change request, just a proposal: I think it would be an improvement if we turn this into an object. Now every client of the scheduler has pack/unpack this tuple in order to interact with the scheduler.

Reply:
Probably a good idea. I'll apply this change.